executor/aggfuncs/aggfuncs.go

```diff
@@ -115,7 +123,18 @@ type baseAggFunc struct {
 	// call arg.EvalXXX to get the actual input data for this function.
 	args []expression.Expression
 
-	// ordinal stores the ordinal of the columns in the output chunk, which is
-	// used to append the final result of this function.
-	ordinal int
+	// ordinal[0] stores the ordinal of the columns in the final output chunk, which is
+	// used to append the final result of this aggregate function.
+	// ordinal[1] stores the start ordinal of the columns in the partial output chunk
```

Comment:
We can store all the column ordinals in this slice. Thus we don't need to assume that all the partial results are stored consecutively.

Reply:
The `ordinal` slice is just used to store the result of the aggregate function:
- result `0` is stored in the `ordinal[0]`-th column
- result `1` is stored in the `ordinal[1]`-th column

Reply:
Maybe we need it to be consecutive.
For agg func in finalMode,
we need to read data from row, whose start index is recorded by ordinal[1];
and we also need to Append the final result to the specific index of chk which is recorded by ordinal[0].

Reply:
for now, we only read data from `args`, and write data to columns specified by `ordinal`. I think we don't need to concern whether the input or output data is consecutively stored.

Reply:
After we support new framework for hash agg,
when parallelly executing, consider Partial1Mode --> FinalMode.

We need to store intermediateData in partial workers and pass it to final workers,
intermediateData will be stored in a chunk, and the start colIdx is specified by ordinal[1],
so we need it to be consecutive.